src/replication.c

```diff
@@ -1708,15 +1709,10 @@ void rdbPipeReadHandler(struct aeEventLoop *eventLoop, int fd, void *clientData,
                 server.rdb_pipe_numconns_writing++;
                 connSetWriteHandler(conn, rdbPipeWriteHandler);
             }
-            stillAlive++;
         }
 
-        if (stillAlive == 0) {
-            serverLog(LL_WARNING, "Diskless rdb transfer, last replica dropped, killing fork child.");
-            killRDBChild();
-        }
         /*  Remove the pipe read handler if at least one write handler was set. */
-        if (server.rdb_pipe_numconns_writing || stillAlive == 0) {
+        if (server.rdb_pipe_numconns_writing) {
```

Comment:
Just so that I understand: in case we kill the RDB child when the last replica unlinks, we will either read '0' before in rdbPipeReadHandler or we will read data from the closed pipe but  since we will not write the data to any replica, we will just remove the read handler right? However this will not call  rdbPipeReadHandler again which will close the read pipe end. Maybe I got it all wrong :) @enjoy-binbin did I?

Reply:
Oh forgot we are reading in a loop here :) sorry then it looks fine to me.